### PR TITLE
feat(api-ci-ui): Integrate Weaviate-Powered Recommendations & Enhance CI/CD Pipeline

### PR DESCRIPTION
**Pull request by hoangsonww**
## Description

This pull request integrates the Weaviate-backed recommendation service end-to-end into our Fusion Electronics backend and CI/CD pipeline. It includes:

* A new `/api/products/recommendations` endpoint that takes an array of product IDs, computes a centroid vector, and returns top nearest-neighbor recommendations from Weaviate.
* MongoDB schema update to store `weaviateId` on each product and two sync scripts:

  * One to upsert all products into Weaviate with deterministic UUIDs and embeddings.
  * One to backfill `weaviateId` in Mongo from existing Weaviate objects.
* Frontend changes to track “visitedProducts” in localStorage (up to 10 most recent), call the new recommendations endpoint, and display paginated recommendations.
* CI/CD pipeline updates:

  * **Recommendation-service sanity tests** in GitHub Actions, using small similarity matrices to smoke-test the recommendations logic.
  * Inclusion of Weaviate sync and recommendation jobs in the overall matrix.
  * Final pipeline summary step prints out status of formatting, backend/frontend tests, recommendation scenarios, and Docker image builds.

Fixes #456 (Weaviate recommendation integration)

## Type of change

* [x] New feature
* [ ] Bug fix
* [ ] Breaking change
* [ ] Documentation update

## How Has This Been Tested?

* **Unit tests** for the new recommendation endpoint, including edge cases (empty input, no vectors, mixed IDs).
* **Integration tests** against a local Weaviate instance with sample products to verify correct nearest-neighbor results.
* **Manual testing** in the browser: visiting products adds to localStorage, recommendations appear and paginate correctly.
* **CI pipeline** runs formatting, backend tests, frontend tests, recommendation sanity scenarios (cold-start, diversity, performance matrices), and Docker image builds without errors.

## Checklist

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in complex vector-sync and recommendation logic
* [x] I have made corresponding changes to the documentation (updated API docs and README with recommendation usage)
* [x] My changes generate no new warnings
* [x] I have added tests that prove my feature works (unit + integration)
* [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -5,12 +5,16 @@
...
-    "start": "nodemon --exec 'cd seed && node productSeeds.js dev && cd .. && node index.js'",
-    "build": "cd seed && node productSeeds.js dev && cd .. && node index.js",
+    "start": "nodemon --exec 'npm run sync-weaviate && node index.js'",
+    "build": "node index.js",
     "test": "jest",
     "test:watch": "jest --watch",
     "test:coverage": "jest --coverage",
-    "dev": "nodemon --exec 'cd seed && node productSeeds.js dev && cd .. && node index.js'"
+    "seed": "node seed/productSeeds.js dev",
+    "dev": "nodemon --exec 'npm sync-weaviate && node index.js'",
```
[the remark this answers is not in the document]
Comment (hoangsonww) [reply backend/package.json:14]: ok
